### PR TITLE
Feature#17 profile database structure

### LINKED ISSUE #17: Profile Database Structure
# REQUIREMENTS

- Profile:
  - City, state, country
  - Number-phone, description
  - Foreign-key: users_id
- Skills:
  - Name
  -  Foreign-key: users_id
- Social Network:
  -  Twitter, Facebook, LinkendIn, Instagram
  - Foreign-key: users_id

# Acceptance Criteria

- [ ] Saving  all data
- [ ] Foreign keys working

### PR DESCRIPTION
**Pull request by DerekAyala**
# Description
What did you implemented/Why did you implemented this:

- Database Profiles , Skills and Social networks created
- Models Profile , Skill and Social network created
- Association Profile , Skill and social network with User
- Profiles , Skills and Social networks have a foreign key to Users

### CODE REVIEW

File: db/migrate/20220203052500_create_social_networks.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateSocialNetworks < ActiveRecord::Migration[6.1]
+  def change
+    create_table :social_networks do |t|
+      t.string :twitter
+      t.string :Facebook
+      t.string :LinkendIn
+      t.string :Instagram
```
Comment (YilKanda) [inline db/migrate/20220203052500_create_social_networks.rb:7]: What is the use of placing the names of social networks here?

Comment (DerekAyala) [reply db/migrate/20220203052500_create_social_networks.rb:7]: They are going to save a link to the social network

**Review by AngelBFdev — APPROVED** [review]
LGTM!

**Review by javs1 — APPROVED** [review]

**Review by JuanequeX — APPROVED** [review]
LGTM

File: db/migrate/20220203052500_create_social_networks.rb (hunk 1 above)
Comment (YilKanda) [reply db/migrate/20220203052500_create_social_networks.rb:7]: I think that the best thing would be to place two fields: social_network_name and url. That way, if you want to add more social networks you can do it in the data and not in the table structure. And fill them with an **[enum](https://naturaily.com/blog/ruby-on-rails-enum)**. What do you think @DerekAyala ?

**Review by YilKanda — COMMENTED** [review]
I think that instead of putting the social networks directly in the CreateSocialNetworks table, it would be better to put two fields: social_network_name and url. That way, if you want to add more social networks, you can do it in the data structure and not in the table. And fill them with an [enum](https://naturaily.com/blog/ruby-on-rails-enum). What do you think @DerekAyala?

**Comment by DerekAyala** [comment]
@YilKanda I choose this way because I can show the social network icon based on the column name if this is not null

**Pull request merged**